cmd/rpc/server.go

```diff
@@ -242,20 +238,18 @@ func (server *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	var callResponse CallResponse
-	var err error
-	callResponse.ReplyBytes, err = server.call(callRequest.Method, callRequest.ArgBytes)
-	if err != nil {
+	callResponse := CallResponse{}
+	buf := bufPool.Get()
+	defer bufPool.Put(buf)
+
+	if err := server.call(callRequest.Method, callRequest.ArgBytes, buf); err != nil {
 		callResponse.Error = err.Error()
 	}
+	callResponse.ReplyBytes = buf.Bytes()
 
-	data, err := gobEncode(callResponse)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		return
-	}
+	gob.NewEncoder(w).Encode(callResponse)
 
-	w.Write(data)
+	w.(http.Flusher).Flush()
```

Comment:
shouldn't we make sure here that the writer supports flushing?

Reply:
HTTP response writer always supports flushing we don't need to validate if it supports or not.